CoreScriptsRoot/Modules/VR/NotificationHub.lua

```diff
@@ -0,0 +1,713 @@
+NotificationHubModule.VRIsExclusive = true
+NotificationHubModule.VRClosesNonExclusive = true
+NotificationHubModule.UnreadCountChanged = function() end
+VRHub:RegisterModule(NotificationHubModule)
+
+local notificationsPanel = Panel3D.Get("Notifications")
+local notificationsWindow = nil
+local detailsPanel = Panel3D.Get("NotificationDetails")
+local detailsWindow = nil
+
+local WindowFrame = {} do
+	local windows = {}
+	local WindowFrame_mt = { __index = WindowFrame }
+	function WindowFrame.new(panel, parent, title)
+		local instance = {}
+		table.insert(windows, instance)
+		instance.zeroCF = panel.localCF
+		instance.zOffset = 0
+		instance.desiredZOffset = 0
+		instance.isAnimating = false
+		instance.animateEndZOffset = 0
+		instance.panel = panel
+		instance.panel.OnMouseEnter = function()
+			if not instance.isAnimating then
+				for i, v in pairs(windows) do
+					if not v.isAnimating then
+						v.desiredZOffset = 0
+					end
+				end
+				instance.desiredZOffset = POPOUT_DISTANCE
+			end
+		end
+
+		instance.titlebar = Util:Create "ImageLabel" {
+			Parent = parent,
+
+			Position = UDim2.new(0, -1, 0, -1),
+			Size = UDim2.new(1, 2, 0, WINDOW_TITLEBAR_HEIGHT + 2),
+
+			BackgroundTransparency = 1,
+
+			Image = "rbxasset://textures/ui/VR/rectBackgroundWhite.png",
+			ImageColor3 = BLURRED_TITLEBAR_COLOR,
+			ScaleType = Enum.ScaleType.Slice,
+			SliceCenter = Rect.new(10, 10, 10, 10)
+		}
+		instance.titleText = Util:Create "TextLabel" {
+			Parent = instance.titlebar,
+
+			Position = UDim2.new(0, 1, 0, 1),
+			Size = UDim2.new(1, -2, 1, -2),
+
+			Text = title,
+			TextColor3 = TEXT_COLOR,
+			Font = Enum.Font.SourceSans,
+			FontSize = Enum.FontSize.Size36,
+
+			BackgroundTransparency = 1
+		}
+
+		instance.content = Util:Create "ImageLabel" {
+			Parent = parent,
+
+			Position = UDim2.new(0, -1, 0, WINDOW_TITLEBAR_HEIGHT + 2),
+			Size = UDim2.new(1, 2, 1, -WINDOW_TITLEBAR_HEIGHT - 4),
+
+			BackgroundTransparency = 1,
+
+			Image = "rbxasset://textures/ui/VR/rectBackgroundWhite.png",
+			ImageColor3 = WINDOW_BG_COLOR,
+			ImageTransparency = WINDOW_BG_TRANSPARENCY,
+			ScaleType = Enum.ScaleType.Slice,
+			SliceCenter = Rect.new(10, 10, 10, 10)
+		}	
+
+		return setmetatable(instance, WindowFrame_mt)
+	end
+
+	function WindowFrame:SetTitle(title)
+		self.titleText.Text = title
+	end
+
+	function WindowFrame:AddCloseButton(callback)
+		local closeBtnSize = 48
+		local closeBtnOffset = 14
+		self.closeButton = Util:Create "ImageButton" {
+			Parent = self.titlebar,
+
+			Position = UDim2.new(0, closeBtnOffset, 0, closeBtnOffset),
+			Size = UDim2.new(0, closeBtnSize, 0, closeBtnSize),
+
+			BackgroundTransparency = 1, 
+
+			Image = "rbxasset://textures/ui/VR/closeButtonPadded.png"
+		}
+		self.closeButton.MouseButton1Click:connect(callback)
+	end
+
+	function WindowFrame:AnimateOut()
+		self.isAnimating = true
+		self.desiredZOffset = ANIMATE_OUT_DISTANCE * 2
+		self.animateEndZOffset = ANIMATE_OUT_DISTANCE
```

Comment:
Aha, I meant to revisit this. It looked really ugly if it disappeared at the end of its tween. Do we have a property tweening function anywhere?